addons/mail/models/mail_thread_main_attachment.py

```diff
@@ -21,11 +21,21 @@ def _message_post_after_hook(self, message, msg_values):
             for attachment_command in (msg_values['attachment_ids'] or [])
         ])
 
-    def _message_set_main_attachment_id(self, attachment_ids):
-        if attachment_ids and not self.message_main_attachment_id:
+    def _message_set_main_attachment_id(self, attachment_ids, force=False, filter_xml=True):
+        """ Update 'main' attachment.
+
+        :param list attachment_ids: new main attachment IDS; if several attachments
+          are given, we search for pdf or image first;
+        :param boolean force: if set, replace an existing attachment; otherwise
```

Comment:
Missing docstring for "filter_xml".

(Which filters slightly more than xml, but not sure how to find a good name for it ^^)